--- hdf5-1.8.1/config/gnu-flags.orig	2008-05-29 18:39:02.000000000 +0200
+++ hdf5-1.8.1/config/gnu-flags	2008-06-04 22:33:32.839106766 +0200
@@ -143,7 +143,6 @@
     # Production
     case "$cc_vendor-$cc_version" in
       gcc-2.95.[34])
-        PROD_CFLAGS="-O3"
         ;;
       gcc-3.[0-4]*|gcc-4.[0123]*)
         # The optimization level is reduced for gcc 3.* and 4.* due to problems
@@ -151,14 +150,11 @@
         # optimization levels (which shows up as failures for various integer
         # types -> long long conversions in the test/dtypes test).  Perhaps
         # later versions of gcc will fix this bug... - QAK - 2003/10/20
-        PROD_CFLAGS="-O"
         ;;
       gcc-4*)
         # Be optimistic about future versions of gcc.. :-) - QAK - 2003/10/20
-        PROD_CFLAGS="-O3"
         ;;
       *)
-        PROD_CFLAGS="-O"
         ;;
     esac
 
