--- hdf5-1.10.9/configure.ac.orig	2023-02-04 17:50:31.414727750 +0100
+++ hdf5-1.10.9/configure.ac	2023-02-04 17:56:08.986232301 +0100
@@ -2146,12 +2146,17 @@ AC_SEARCH_LIBS([clock_gettime], [rt posi
 AC_CHECK_FUNCS([alarm asprintf clock_gettime difftime fcntl flock fork frexpf])
 AC_CHECK_FUNCS([frexpl gethostname getrusage gettimeofday])
 AC_CHECK_FUNCS([lstat rand_r random setsysinfo])
-AC_CHECK_FUNCS([signal longjmp setjmp siglongjmp sigsetjmp sigprocmask])
+AC_CHECK_FUNCS([signal longjmp setjmp siglongjmp sigprocmask])
 AC_CHECK_FUNCS([snprintf srandom strdup symlink system])
 AC_CHECK_FUNCS([strtoll strtoull])
 AC_CHECK_FUNCS([tmpfile vasprintf vsnprintf waitpid])
 AC_CHECK_FUNCS([roundf lroundf llroundf round lround llround])
 
+dnl cannot be detected in glibc by plain AC_CHECK_FUNCS
+AH_TEMPLATE([HAVE_SIGSETJMP], [Have sigsetjmp function])
+AC_CHECK_FUNC([sigsetjmp], [AC_DEFINE([HAVE_SIGSETJMP])],
+	[AC_CHECK_FUNC([__sigsetjmp], [AC_DEFINE([HAVE_SIGSETJMP])])])
+
 ## ----------------------------------------------------------------------
 ## Check compiler characteristics
 ##
