diff -Nur hdf5-1.4.2-patch1.orig/configure.in hdf5-1.4.2-patch1/configure.in
--- hdf5-1.4.2-patch1.orig/configure.in	Fri Jul 20 19:41:09 2001
+++ hdf5-1.4.2-patch1/configure.in	Fri Oct 26 00:27:16 2001
@@ -577,13 +577,16 @@
 AC_SUBST(TESTH4TOH5) TESTH4TOH5='$(srcdir)/testh4toh5.sh'
 
 AC_ARG_WITH(hdf4,[  --with-hdf4=INC,LIB     Use the HDF4 library],,withval=no)
+H4LIBS=""
 case "$withval" in
    yes)
+      save_LIBS="$LIBS"
       AC_CHECK_HEADERS(mfhdf.h,,unset H5TOH4 TESTH5TOH4 H4TOH5 H4TOH5TEST TESTH4TOH5)
-      AC_CHECK_LIB(z,compress,,unset H5TOH4 TESTH5TOH4 H4TOH5 H4TOH5TEST TESTH4TOH5)
-      AC_CHECK_LIB(jpeg,jpeg_start_compress,,unset H5TOH4 TESTH5TOH4 H4TOH5 H4TOH5TEST TESTH4TOH5)
-      AC_CHECK_LIB(df,Hstartaccess,,unset H5TOH4 TESTH5TOH4 H4TOH5 H4TOH5TEST TESTH4TOH5)
-      AC_CHECK_LIB(mfhdf,SDstart,,unset H5TOH4 TESTH5TOH4 H4TOH5 H4TOH5TEST TESTH4TOH5)
+      AC_CHECK_LIB(z,compress,H4LIBS="-lz",unset H5TOH4 TESTH5TOH4 H4TOH5 H4TOH5TEST TESTH4TOH5)
+      AC_CHECK_LIB(jpeg,jpeg_start_compress,H4LIBS="$H4LIBS -ljpeg",unset H5TOH4 TESTH5TOH4 H4TOH5 H4TOH5TEST TESTH4TOH5)
+      AC_CHECK_LIB(df,Hstartaccess,H4LIBS="$H4LIBS -ldf",unset H5TOH4 TESTH5TOH4 H4TOH5 H4TOH5TEST TESTH4TOH5)
+      AC_CHECK_LIB(mfhdf,SDstart,H4LIBS="$H4LIBS -lmfhdf",unset H5TOH4 TESTH5TOH4 H4TOH5 H4TOH5TEST TESTH4TOH5)
+      LIBS="$save_LIBS"
       ;;
    no)
       AC_MSG_CHECKING(for HDF4)
@@ -606,18 +609,23 @@
       if test -n "$hdf4_lib"; then
          saved_LDFLAGS="$LDFLAGS"
          LDFLAGS="$LDFLAGS -L$hdf4_lib"
-         AC_CHECK_LIB(z,compress,,unset H5TOH4 TESTH5TOH4 H4TOH5 H4TOH5TEST TESTH4TOH5)
-         AC_CHECK_LIB(jpeg,jpeg_start_compress,,unset H5TOH4 TESTH5TOH4 H4TOH5 H4TOH5TEST TESTH4TOH5)
-         AC_CHECK_LIB(df,Hstartaccess,,unset H5TOH4 TESTH5TOH4 H4TOH5 H4TOH5TEST TESTH4TOH5)
-         AC_CHECK_LIB(mfhdf,SDstart,,unset H5TOH4 TESTH5TOH4 H4TOH5 H4TOH5TEST TESTH4TOH5)
+	  save_LIBS="$LIBS"
+         AC_CHECK_LIB(z,compress,H4LIBS="-lz",unset H5TOH4 TESTH5TOH4 H4TOH5 H4TOH5TEST TESTH4TOH5)
+         AC_CHECK_LIB(jpeg,jpeg_start_compress,H4LIBS="$H4LIBS -ljpeg",unset H5TOH4 TESTH5TOH4 H4TOH5 H4TOH5TEST TESTH4TOH5)
+         AC_CHECK_LIB(df,Hstartaccess,H4LIBS="$H4LIBS -ldf",unset H5TOH4 TESTH5TOH4 H4TOH5 H4TOH5TEST TESTH4TOH5)
+         AC_CHECK_LIB(mfhdf,SDstart,H4LIBS="$H4LIBS -lmfhdf",unset H5TOH4 TESTH5TOH4 H4TOH5 H4TOH5TEST TESTH4TOH5)
+	  LIBS="$save_LIBS"
       else
-         AC_CHECK_LIB(z,compress,,unset H5TOH4 TESTH5TOH4 H4TOH5 H4TOH5TEST TESTH4TOH5)
-         AC_CHECK_LIB(jpeg,jpeg_start_compress,,unset H5TOH4 TESTH5TOH4 H4TOH5 H4TOH5TEST TESTH4TOH5)
-         AC_CHECK_LIB(df,Hstartaccess)
-         AC_CHECK_LIB(mfhdf,SDstart)
+	  save_LIBS="$LIBS"
+         AC_CHECK_LIB(z,compress,H4LIBS="-lz",unset H5TOH4 TESTH5TOH4 H4TOH5 H4TOH5TEST TESTH4TOH5)
+         AC_CHECK_LIB(jpeg,jpeg_start_compress,H4LIBS="$H4LIBS -ljpeg",unset H5TOH4 TESTH5TOH4 H4TOH5 H4TOH5TEST TESTH4TOH5)
+         AC_CHECK_LIB(df,Hstartaccess,H4LIBS="$H4LIBS -ldf")
+         AC_CHECK_LIB(mfhdf,SDstart,H4LIBS="$H4LIBS -lmfhdf")
+	  LIBS="$save_LIBS"
       fi
       ;;
 esac
+AC_SUBST(H4LIBS)
 
 dnl ----------------------------------------------------------------------
 dnl Is the GNU zlib present?  It has a header file `zlib.h' and a library
diff -Nur hdf5-1.4.2-patch1.orig/tools/h4toh5/Makefile.in hdf5-1.4.2-patch1/tools/h4toh5/Makefile.in
--- hdf5-1.4.2-patch1.orig/tools/h4toh5/Makefile.in	Thu Mar 22 01:00:33 2001
+++ hdf5-1.4.2-patch1/tools/h4toh5/Makefile.in	Fri Oct 26 00:20:41 2001
@@ -52,9 +52,9 @@
 $(PROGS): $(LIBTOOLS) $(LIBHDF5)
 
 h4toh5test: h4toh5test.lo
-	@$(LT_LINK_EXE) $(CFLAGS) -o $@ h4toh5test.lo $(LIBTOOLS) $(LIBHDF5) $(LDFLAGS) $(LIBS)
+	@$(LT_LINK_EXE) $(CFLAGS) -o $@ h4toh5test.lo $(LIBTOOLS) $(LIBHDF5) $(LDFLAGS) $(LIBS) @H4LIBS@
 
 h4toh5: $(PROG_OBJ)
-	@$(LT_LINK_EXE) $(CFLAGS) -o $@ $(PROG_OBJ) $(LIBTOOLS) $(LIBHDF5) $(LDFLAGS) $(LIBS)
+	@$(LT_LINK_EXE) $(CFLAGS) -o $@ $(PROG_OBJ) $(LIBTOOLS) $(LIBHDF5) $(LDFLAGS) $(LIBS) @H4LIBS@
 
 @CONCLUDE@
diff -Nur hdf5-1.4.2-patch1.orig/tools/h5toh4/Makefile.in hdf5-1.4.2-patch1/tools/h5toh4/Makefile.in
--- hdf5-1.4.2-patch1.orig/tools/h5toh4/Makefile.in	Thu Mar 22 01:00:39 2001
+++ hdf5-1.4.2-patch1/tools/h5toh4/Makefile.in	Fri Oct 26 00:21:13 2001
@@ -55,6 +55,6 @@
 $(PROGS): $(LIBTOOLS) $(LIBHDF5)
 
 h5toh4: h5toh4.lo
-	@$(LT_LINK_EXE) $(CFLAGS) -o $@ h5toh4.lo $(LIBTOOLS) $(LIBHDF5) $(LDFLAGS) $(LIBS)
+	@$(LT_LINK_EXE) $(CFLAGS) -o $@ h5toh4.lo $(LIBTOOLS) $(LIBHDF5) $(LDFLAGS) $(LIBS) @H4LIBS@
 
 @CONCLUDE@
