--- hdf5-1.4.5/configure.in.wiget	Mon Jan 27 18:45:56 2003
+++ hdf5-1.4.5/configure.in	Tue Apr  1 18:16:48 2003
@@ -562,9 +562,9 @@
 esac
 
 AC_TRY_COMPILE([#include <sys/types.h>],
-               [off64_t n = 0;],
+               [off64_t n = 0;],[
                AC_CHECK_FUNCS(lseek64 fseek64),
-               AC_MSG_RESULT([skipping test for lseek64() and fseek64()]))
+               AC_MSG_RESULT([skipping test for lseek64() and fseek64()])])
 
 dnl ----------------------------------------------------------------------
 dnl Data types and their sizes.
