--- hdf5-1.10.10/m4/aclocal_cxx.m4.orig	2023-08-21 06:23:45.185096953 +0200
+++ hdf5-1.10.10/m4/aclocal_cxx.m4	2023-08-21 06:24:56.672450942 +0200
@@ -26,7 +26,7 @@ dnl have a [PAC] prefix.
 dnl Checking if C++ needs old style header files in includes
 AC_DEFUN([PAC_PROG_CXX_HEADERS],[
   AC_MSG_CHECKING([if $CXX needs old style header files in includes])
-  TEST_SRC="`(echo \"#define OLD_HEADER_FILENAME 1\"; cat $srcdir/config/cmake/HDFCXXTests.cpp)`"
+  TEST_SRC="$(echo "#define OLD_HEADER_FILENAME 1"; cat $srcdir/config/cmake/HDFCXXTests.cpp)"
 
   AC_LINK_IFELSE([AC_LANG_SOURCE([$TEST_SRC])],
     [AC_MSG_RESULT([no])],
@@ -38,7 +38,7 @@ AC_DEFUN([PAC_PROG_CXX_HEADERS],[
 dnl Checking if ++ can handle namespaces
 AC_DEFUN([PAC_PROG_CXX_NAMESPACE],[
   AC_MSG_CHECKING([if $CXX can handle namespaces])
-  TEST_SRC="`(echo \"#define HDF_NO_NAMESPACE 1\"; cat $srcdir/config/cmake/HDFCXXTests.cpp)`"
+  TEST_SRC="$(echo "#define HDF_NO_NAMESPACE 1"; cat $srcdir/config/cmake/HDFCXXTests.cpp)"
 
   AC_LINK_IFELSE([AC_LANG_SOURCE([$TEST_SRC])], [AC_MSG_RESULT([yes])],
      [AC_MSG_RESULT([no])
@@ -49,7 +49,7 @@ AC_DEFUN([PAC_PROG_CXX_NAMESPACE],[
 dnl Checking if C++ supports std
 AC_DEFUN([PAC_PROG_CXX_STD],[
   AC_MSG_CHECKING([if $CXX supports std])
-  TEST_SRC="`(echo \"#define HDF_NO_STD 1\"; cat $srcdir/config/cmake/HDFCXXTests.cpp)`"
+  TEST_SRC="$(echo "#define HDF_NO_STD 1"; cat $srcdir/config/cmake/HDFCXXTests.cpp)"
 
   AC_LINK_IFELSE([AC_LANG_SOURCE([$TEST_SRC])], [AC_MSG_RESULT([yes])],
      [AC_MSG_RESULT([no])
@@ -60,7 +60,7 @@ AC_DEFUN([PAC_PROG_CXX_STD],[
 dnl Checking if C++ has offsetof extension
 AC_DEFUN([PAC_PROG_CXX_OFFSETOF],[
   AC_MSG_CHECKING([if $CXX has offsetof extension])
-  TEST_SRC="`(echo \"#define CXX_HAVE_OFFSETOF 1\"; cat $srcdir/config/cmake/HDFCXXTests.cpp)`"
+  TEST_SRC="$(echo "#define CXX_HAVE_OFFSETOF 1"; cat $srcdir/config/cmake/HDFCXXTests.cpp)"
 
   AC_LINK_IFELSE([AC_LANG_SOURCE([$TEST_SRC])],[AC_MSG_RESULT([yes])
     AC_DEFINE([CXX_HAVE_OFFSETOF], [1], [Define if C++ compiler recognizes offsetof])],
@@ -70,7 +70,7 @@ AC_DEFUN([PAC_PROG_CXX_OFFSETOF],[
 dnl Checking if C++ can handle static cast
 AC_DEFUN([PAC_PROG_CXX_STATIC_CAST],[
   AC_MSG_CHECKING([if $CXX can handle static cast])
-  TEST_SRC="`(echo \"#define NO_STATIC_CAST 1\"; cat $srcdir/config/cmake/HDFCXXTests.cpp)`"
+  TEST_SRC="$(echo "#define NO_STATIC_CAST 1"; cat $srcdir/config/cmake/HDFCXXTests.cpp)"
 
   AC_LINK_IFELSE([AC_LANG_SOURCE([$TEST_SRC])], [AC_MSG_RESULT([yes])],
     [AC_MSG_RESULT([no])
