--- hdf5-1.8.14/configure.ac.orig	2014-11-14 21:04:02.769225653 +0100
+++ hdf5-1.8.14/configure.ac	2014-11-14 21:06:03.502553920 +0100
@@ -2253,10 +2253,15 @@
 AC_CHECK_FUNCS([alarm BSDgettimeofday fork frexpf frexpl])
 AC_CHECK_FUNCS([gethostname getpwuid getrusage lstat])
 AC_CHECK_FUNCS([rand_r random setsysinfo])
-AC_CHECK_FUNCS([signal longjmp setjmp siglongjmp sigsetjmp sigprocmask])
+AC_CHECK_FUNCS([signal longjmp setjmp siglongjmp sigprocmask])
 AC_CHECK_FUNCS([snprintf srandom strdup symlink system])
 AC_CHECK_FUNCS([tmpfile asprintf vasprintf waitpid])
 
+dnl cannot be detected in glibc by plain AC_CHECK_FUNCS
+AH_TEMPLATE([HAVE_SIGSETJMP], [Have sigsetjmp function])
+AC_CHECK_FUNC([sigsetjmp], [AC_DEFINE([HAVE_SIGSETJMP])],
+	[AC_CHECK_FUNC([__sigsetjmp], [AC_DEFINE([HAVE_SIGSETJMP])])])
+
 ## Check for vsnprintf() separately, so we can detect situations where it
 ## doesn't return the correct size for formatted strings that are too large
 ## for the buffer provided
