--- hdf5-1.8.16/config/gnu-flags.orig	2015-11-26 21:06:26.741682874 +0100
+++ hdf5-1.8.16/config/gnu-flags	2015-11-26 21:08:37.605010716 +0100
@@ -99,10 +99,10 @@
     # Production
     case "$cc_vendor-$cc_version" in
       gcc-[34].*)
-        PROD_CFLAGS="-O3"
+        PROD_CFLAGS=
         ;;
       gcc-5.*)
-        PROD_CFLAGS="-O3 -fstdarg-opt"
+        PROD_CFLAGS="-fstdarg-opt"
         ;;
       *)
         PROD_CFLAGS="-O -finline-functions"
